src/cpp/core/r_util/RSourceIndex.cpp

```diff
@@ -366,6 +366,58 @@ void libraryCallIndexer(const RTokenCursor& cursor,
    }
 }
 
+void testThatCallIndexer(const RTokenCursor& cursor,
+                         const IndexStatus& status,
+                         bool isReadOnlyFile, 
+                         RSourceIndex* pIndex)
+{
+   if (!cursor.isType(RToken::ID))
+      return;
+   
+   if (!(cursor.contentEquals(L"test_that")))
+      return;
+   
+   RTokenCursor clone = cursor.clone();
+   if (!clone.moveToNextSignificantToken())
+      return;
+   
+   if (!clone.isType(RToken::LPAREN))
+      return;
+   
+   if (!clone.moveToNextSignificantToken())
+      return;
+   
+   if (clone.isType(RToken::STRING))
+   {
+      static const boost::regex removeQuotesRegex("^['\"](.*)['\"]$");
+      static const boost::regex removeRawStringQuotesRegex("^r['\"][(](.*)[)]['\"]$");
```

Comment:
Raw strings in R are very permissive; for example:

```
> R"(abc)"
[1] "abc"
> R"{abc}"
[1] "abc"
> R"[abc]"
[1] "abc"
> R"--[abc]--"
[1] "abc"
```

That is, you can use `r` or `R` to begin a raw string, a variable number of `-` separators, and also your own choice of `(`, `{` or `[`. See for example:

https://github.com/rstudio/rstudio/blob/50af541be4241393554976378005e876285d3230/src/cpp/core/r_util/RTokenizer.cpp#L248-L353

Rather than a regular expression, because we know the token must already be a string, I think you should do something like the following:

1. Check for a leading `r` or `R` for a raw string versus a "regular" string,
2. Find the index of the first character of `"({["`,
3. Use that to extract the inner substring of the raw string, e.g. `content.substr(<start>, <end>)` where both `<start>` and `<end>` are computed based on the size of the raw string prefix.

In practice I doubt any user is going to use raw strings in this context, and even if they do the worst case is they see the raw string as-is in the UI, so that's probably not the end of the world.

Reply:
Thanks. I'll forget about this until someone someday uses a raw string in a `test_that()` description.